### PR TITLE
[Merged by Bors] - feat: product of graphs of functions

### PR DESCRIPTION
**Pull request by YaelDillies**
From PFR

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Function.lean [hunk 1]
```diff
@@ -688,6 +688,10 @@ lemma exists_eq_graphOn [Nonempty β] {s : Set (α × β)} :
   .trans ⟨fun ⟨f, t, hs⟩ ↦ ⟨f, by rw [hs, image_fst_graphOn]⟩, fun ⟨f, hf⟩ ↦ ⟨f, _, hf⟩⟩
     exists_eq_graphOn_image_fst
 
+lemma graphOn_prod_graphOn (s : Set α) (t : Set β) (f : α → γ) (g : β → δ) :
+    s.graphOn f ×ˢ t.graphOn g = Equiv.prodProdProdComm .. ⁻¹' (s ×ˢ t).graphOn (Prod.map f g) := by
```
Comment (b-mehta) [inline Mathlib/Data/Set/Function.lean:692]: Is there a reason to prefer the preimage over the image spelling on the rhs?

File: Mathlib/Data/Set/Function.lean [hunk 2]
```diff
@@ -688,6 +688,10 @@ lemma exists_eq_graphOn [Nonempty β] {s : Set (α × β)} :
...
+  aesop
+
```
Comment (b-mehta) [inline Mathlib/Data/Set/Function.lean:694]:
```suggestion

lemma prod_graphOn_map (s : Set α) (t : Set β) (f : α → γ) (g : β → δ) :
    (s ×ˢ t).graphOn (Prod.map f g) = Equiv.prodProdProdComm .. ⁻¹' s.graphOn f ×ˢ t.graphOn g := by
  aesop
```
It seems helpful to go the other way round too.

File: Mathlib/Data/Set/Function.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Set/Function.lean:692]: Preimages along equivs are usually preferred to images due to better defeqs

File: Mathlib/Data/Set/Function.lean [hunk 3]
```diff
@@ -688,6 +688,13 @@ lemma exists_eq_graphOn [Nonempty β] {s : Set (α × β)} :
...
+lemma graphOn_prod_prodMap (s : Set α) (t : Set β) (f : α → γ) (g : β → δ) :
+    (s ×ˢ t).graphOn (Prod.map f g) = Equiv.prodProdProdComm .. ⁻¹' s.graphOn f ×ˢ t.graphOn g := by
+  aesop
```
Comment (YaelDillies) [inline Mathlib/Data/Set/Function.lean:697]:
```suggestion
  aesop

```

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**